src/components/IndexTable/IndexTable.tsx

```diff
@@ -118,21 +125,57 @@ function IndexTableBase({
     );
   }, [handleSelectionChange, selectedItemsCount]);
 
-  const resizeTableHeadings = useCallback(() => {
-    if (!tableElement.current || !scrollableContainerElement.current) {
-      return;
-    }
+  const calculateFirstHeaderOffset = useCallback(() => {
+    return condensed
+      ? tableHeadingRects.current[0].offsetWidth
+      : tableHeadingRects.current[0].offsetWidth +
+          tableHeadingRects.current[1].offsetWidth;
+  }, [condensed]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeTableHeadings = useCallback(
+    throttle(() => {
+      if (!tableElement.current || !scrollableContainerElement.current) {
+        return;
+      }
 
-    const measuredTableHeadingRects = Array.from(
-      tableElement.current.querySelectorAll('[data-index-table-heading]'),
-    ).map((heading) => ({
-      offsetWidth: heading instanceof HTMLElement ? heading.offsetWidth : 0,
-      offsetLeft: heading instanceof HTMLElement ? heading.offsetLeft : 0,
-    }));
-    const boundingRect = scrollableContainerElement.current.getBoundingClientRect();
-    tablePosition.current = {top: boundingRect.top, left: boundingRect.left};
-    tableHeadingRects.current = measuredTableHeadingRects;
-  }, []);
+      const boundingRect = scrollableContainerElement.current.getBoundingClientRect();
+      tablePosition.current = {top: boundingRect.top, left: boundingRect.left};
+
+      tableHeadingRects.current = tableHeadings.current.map((heading) => ({
+        offsetWidth: heading.offsetWidth || 0,
+        offsetLeft: heading.offsetLeft || 0,
+      }));
+
+      if (tableHeadings.current.length === 0) {
+        return;
+      }
+
+      // update left offset for first column
+      if (tableHeadings.current.length > 1)
+        tableHeadings.current[1].style.left = `${tableHeadingRects.current[0].offsetWidth}px`;
```

Comment:
have you considered moving these inline styles to the css file, with css vars for the necessary data?

Reply:
I didn't! What benefit does this give us (besides making it so we're less handsy with setting style props)? Might prevent some layout shifts I guess?

Reply:
depending on implementation, it can prevent some layout shifts. basically, instead of setting inline styles on several elements (which can cause a layout) you set inline css vars on 1 common parent element, causing 1 layout.

it also moves the presentation logic into css, where it belongs. JS is only responsible then for providing data that css needs.

it also makes it easier/possible to override the implementation in either a 3rd party app, and/or via composition

Reply:
Ah yeah that's what I figured. Our issue is that the number of columns is dynamic... which makes solving this non-programmatically difficult. Good insights though!